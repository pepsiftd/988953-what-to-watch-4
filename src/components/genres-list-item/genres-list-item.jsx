import React from 'react';
import PropTypes from 'prop-types';

const GenresListItem = ({genre, titleClickHandler, isActive}) => {
  const activeGenreClass = `catalog__genres-item--active`;

  return (
    <li className={`catalog__genres-item ${isActive ? activeGenreClass : ``}`}>
      <a href="#" className="catalog__genres-link" onClick={(evt) => {
        evt.preventDefault();
        titleClickHandler(genre);
      }}>{genre}</a>
    </li>
  );
};

GenresListItem.propTypes = {
  genre: PropTypes.string.isRequired,
  titleClickHandler: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
};

export {GenresListItem};
