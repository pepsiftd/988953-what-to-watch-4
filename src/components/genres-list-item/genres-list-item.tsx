import React from 'react';

interface Props {
  genre: string;
  onTitleClick: (id: string) => void;
  isActive: boolean;
}

const GenresListItem: React.FunctionComponent<Props> = ({genre, onTitleClick, isActive}) => {
  const activeGenreClass = `catalog__genres-item--active`;

  return (
    <li className={`catalog__genres-item ${isActive ? activeGenreClass : ``}`}>
      <a href="#" className="catalog__genres-link" onClick={(evt) => {
        evt.preventDefault();
        onTitleClick(genre);
      }}>{genre}</a>
    </li>
  );
};

export {GenresListItem};
