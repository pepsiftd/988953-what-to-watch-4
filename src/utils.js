const getUniqueItems = (list) => {
  return list.filter((it, i, array) => {
    return array.indexOf(it) === i;
  });
};

const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const getMoviesByGenre = (movies, genre) => {
  return genre === `All genres` ? movies : movies.filter((movie) => movie.genre === genre);
};

const getGenresFromMovies = (movies) => {
  return getUniqueItems(movies.map((movie) => movie.genre));
};

export {extend, getMoviesByGenre, getUniqueItems, getGenresFromMovies};
