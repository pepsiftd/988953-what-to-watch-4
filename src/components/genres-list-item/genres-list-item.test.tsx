import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {noop} from '@/utils';

import {GenresListItem} from './genres-list-item';

describe(`GenresListItem renders`, () => {
  it(` active Item correctly`, () => {
    const tree = renderer.create(
        <GenresListItem
          genre={`All genres`}
          onTitleClick={noop}
          isActive={true}
        />
    );

    expect(tree).toMatchSnapshot();
  });

  it(` not active Item correctly`, () => {
    const tree = renderer.create(
        <GenresListItem
          genre={`All genres`}
          onTitleClick={noop}
          isActive={false}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
