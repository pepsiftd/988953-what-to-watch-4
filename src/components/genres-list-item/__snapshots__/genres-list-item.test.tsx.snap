// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresListItem renders  active Item correctly 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    All genres
  </a>
</li>
`;

exports[`GenresListItem renders  not active Item correctly 1`] = `
<li
  className="catalog__genres-item "
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    All genres
  </a>
</li>
`;
